### ISSUE TITLE
Noisy learning of Pytorch implementation compared to Tensorflow in homwork 2

### ISSUE THREAD

**Issue by erfanMhi** [erfanMhi/Deep-Reinforcement-Learning-CS285-Pytorch#1]
### Experiments
The following experiments have been conducted using both Tensorflow and Pytorch Impelementaitons:
```sh
python cs285/scripts/run_hw2_policy_gradient.py --env_name CartPole-v1 --exp_name test_pg_cartpole
python cs285/scripts/run_hw2_policy_gradient.py --env_name InvertedPendulum-v2 --exp_name test_pg_pendulum
```
The resulting plots of average return and standard deviation of return in each step for Pytorch implementation of are as follows (Blue line belongs to Inverted-pendulum and orange line belongs to Cart-pole):

![pytorch-average-return](https://user-images.githubusercontent.com/25865840/88482948-d8972d00-cf79-11ea-9f8f-ed1f616d2651.png)

![pytorch-std-return](https://user-images.githubusercontent.com/25865840/88482949-dc2ab400-cf79-11ea-80ce-e60172bc7fb3.png)

On the other hand, the same graphs for the Tensorflow implementation are as follows:

![tensorflow-average-return](https://user-images.githubusercontent.com/25865840/88483074-d08bbd00-cf7a-11ea-96e4-2cd0336d7355.png)

![tensorflow-std-return](https://user-images.githubusercontent.com/25865840/88483078-d2ee1700-cf7a-11ea-9f58-6c02cfa39d47.png)

### Problem Statement
As evident from these graphs, the Pytorch implementation experiences high amounts of fluctuations both in average return and its average std remains high even in the last steps whereas the return of the Tensorflow implementation converges to the global optimum only after a couple of generations and have been exposed to a low amount of noise during training. These findings suggest that there is a problem with Pytorch implementation. However, I checked the weight initialization, loss function, and architecture of both implementations, and they are the same. There should be something different in the Pytorch implementation that I have not considered so far. **Therefore, I call for help from you, dear contributors.**

**Comment by erfanMhi** [comment]
This issue has been resolved in [this commit](https://github.com/erfanMhi/Deep-Reinforcement-Learning-CS285-Pytorch/tree/b197b7d87b5a6a4e16e5b2999efba61499462361).
